apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts

```diff
@@ -0,0 +1,191 @@
+            }
+        });
+        
+        this.componentStore.activeCompany$.pipe(takeUntil(this.destroyed$)).subscribe(activeCompany => {
+            if (activeCompany) {
+                this.activeCompany = activeCompany;
+            }
+        });
+
+        // Subscribe Export Report Success Observable
+        this.componentStore.exportTaxWiseReport$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.downloadReport(response);
+            }
+        });
+
+        // Subscribe Sales Report List Observable
+        this.componentStore.taxWiseReport$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.pagination.page = response?.page;
+                this.pagination.totalItems = response?.totalItems;
+                this.pagination.totalPages = response?.totalPages;
+            }
+        });
+    }
+
+    /**
+    * Get Tax Authority API Call
+    *
+    * @memberof RateWiseReportComponent
+    */
+    public getSalesTaxReport(): void {
+        const formValue = this.salesTaxReportForm.value;
+        if (formValue.taxNumber && formValue.from && formValue.to) {
+            const model: any = {
+                reportType: SalesTaxReport.TaxWise,
+                params: { ...formValue, page: this.pagination.page, count: this.pagination.count },
+                isExport: false
+            };
+            this.componentStore.getSalesTaxReport(model);
+        }
+    }
+
+    /**
+    * Export Tax Authority API Call
+    *
+    * @memberof RateWiseReportComponent
+    */
+    public exportTaxAuthority(): void {
+        const formValue = this.salesTaxReportForm.value;
+        const model: any = {
+            reportType: SalesTaxReport.TaxWise,
+            params: formValue,
+            isExport: true
+        };
+        this.componentStore.exportSalesTaxReport(model);
+    }
+
+
+    /**
+    * This will use for init main formgroup
+    *
+    * @private
+    * @memberof RateWiseReportComponent
+    */
+    private initSalesTaxReportForm(): void {
+        this.salesTaxReportForm = this.formBuilder.group({
+            taxAuthorityUniqueName: [''],
+            branchUniqueName: [''],
+            taxNumber: [''],
+            from: [''],
+            to: ['']
+        });
+    }
+
+    /**
+    * Used to get and set form control value
+    *
+    * @param {string} control
+    * @returns {*}
+    * @memberof RateWiseReportComponent
+    */
+    public getFormControl(control: string): any {
+        return this.salesTaxReportForm.get(control);
+    }
+
+    /**
+     * Download Excel Report
+     *
+     * @param {*} response
+     * @returns {void}
+     * @memberof RateWiseReportComponent
+     */
+    public downloadReport(response: any): void {
+        let blob = this.generalService.base64ToBlob(response, 'application/xls', 512);
+        return saveAs(blob, `Sales Tax Report - Rate wise.csv`);
```

Comment:
When you receive data from the API for the name, make the necessary changes accordingly. Do the same for the code.